cobaya/component.py

```diff
@@ -331,6 +331,8 @@ class CobayaComponent(HasLogger, HasDefaults):
     _at_resume_prefer_new: List[str] = ["version"]
     _at_resume_prefer_old: List[str] = []
 
+    enforce_types: bool = False
```

Comment:
I am not sure about this. Indeed, adding the underscore we cannot enforce types neither through the `yaml` nor through the info dict (as I do in the test for instance). So, I guess this becomes a question of what default behavior we want to use. If the default value of `enforce_types` is `False`, the user can only enforce types on components he/she defines with that flag to `True`. But, for example, types of built-in samplers and such cannot be checked (except if we add in their definition the opposite flag). This may be a good choice since I guess that types of built-in stuff are checked anyway, in a way of the other. Vice versa, with `enforce_types = True` by default, everything will be explicitly checked and there is no need for the user to access that.

Am I missing something?

Reply:
Would probably have to be False by default to keep people from hitting annoying errors and compatibility with existing external likelihoods. Was thinking of this more as an option for likelihood developers who support accurate type hint in their code.

Reply:
Ok great, then I'll make it private and default at `False` :+1: